### DISCUSSION TITLE
Do you use clear and meaningful names when coding?

### DISCUSSION THREAD

**Discussion by giscus** [SSWConsulting/SSW.Rules.Content#9597] [Rules Discussions]
# bfcbc0ef-d2f8-4e77-85d9-97ccb841eb19

Naming things is hard, but also important. Using clear, meaningful names will make your code more readable and reduce cognitive load and risk for your development team.

https://www.ssw.com.au/rules/clear-meaningful-names/

**Comment by JackDevAU** [comment]
Having clear names is massive when working in teams, where there are constant devs coming and going. Being able to quickly make sense of code rapidly speeds up onboarding times. 🚀

**Comment by calumjs** [comment]
Not sure if I agree with the bad example at 3:51 - blanket ban on using "name"
If you're writing object oriented code and the variable is scoped to a class, the class gives context about what the name is.
e.g. product.name tells you it is the name of the product, you don't need to call it product.productName

Sure maybe with users/people it's a little more complex because people have first, last, full names, etc...

**Reply by jakebayliss** [reply]
My opinion was date was the better example of this. Even if Date was scoped to a class, you still need DateCreated, Modified, etc.

**Comment by tiagov8** [comment]
If your code forces others to pause and decipher it, you're already wasting valuable time.

**Comment by jakebayliss** [comment]
I think this rule is the most important of the lot. If a dev has to try to decode your variable names while figuring out what the code is doing then it makes it much more difficult

**Comment by TobyChurches** [comment]
I think @jernejk would agree with me when we say this is a extremely important rule!!! - We have worked on a project where a single entity has fields for `CreatedOn` and `DateCreated` 😬 Even after working on the project for a long time, I still have to check the wiki every time I debug the related feature

**Comment by bradystroud** [comment]
Its important to get this right from the start or you'll find yourself renaming stuff all the time which adds to the confusion.

These days we have autocomplete which means there is no downside to longer names

<img width="408" alt="image" src="https://github.com/user-attachments/assets/d306cac3-41e1-4d74-bec3-d9a6eba4d05e">

**Figure: Cool first letter lookup to save typing long names**

**Comment by jeoffreyfischer** [comment]
Consistency is also important because it directly impacts the readability, maintainability, and overall quality of the codebase

**Comment by awaltos** [comment]
Naming can be difficult especially when the team is large. Following good standards like the ones listed here are important, but also keeping that consistent across the code base helps increase understanding and decrease frustration. OneArea_That_DevsCanGetSuperCreative_And_Verbose_Is_In_UnitTestMethodNames 🤣️

**Comment by StefStar** [comment]
I am not a developer however it always makes more sense to use established clear naming when labelling anything. In hospitality lots of food waste can occur in kitchens if items are not labelled correctly or following a standard ask me how I know....

**Comment by joshbermanssw** [comment]
This is the most important rule in my opinion - shortening or abbreviating always comes back to bite you if you end up trying to look back at code you wrote a while ago and make sense of it... also thoughtful for other devs.

**Comment by JeanThirion** [comment]
Fully disagree with not using "Name" or similar 
It makes a lot of sense to have "Item.name" in a game or in a CMS like SharePoint. 

What's the alternative ? Item.displayName ? No - because it's not necessarily just display.
item.internalName ? Won't work - this implies there's a displayName (which you don't necessarily need)
etc.

Otherwise good video

**Reply by SamBWagner** [reply]
I think `product.name` can, in an of itself, be wholly descriptive.

I think `barcode.name` is bad. `name` is descriptive, only when bestowed context from another source.

The prescription here is to **be as specific as you can be**. In the case of a user, just using `name` is not great because you can be more specific. i.e `user.fullName` or `user.lastName'.

more specificity, leads to a more descriptive name 👍

**Comment by tkapa** [comment]
Names should primarily be consistent. 

Your naming conventions mean nothing if they aren't the same across your project

**Comment by isaaclombardssw** [comment]
Overuse of abbreviations has caused me a lot of pain in the past when trying to understand a piece of code, especially when a lot of seemingly-arbitrary numbers are involved (which ironically is where this happens most).

**Comment by vladislav-kir** [comment]
Meaningful names are very import to make the code easy to understand.

However, I am not 100% agree about using "name" or similar being a bad item.

As a variable name, "name" is not meaningful enough.

But as a property of a Class, I believe it is acceptable. For example, if you have classes Session, Presentation, etc... Them having a consistent property "Name" is better, than each of them having a unique property of that.

**Comment by bettybondoc** [comment]
When I was starting out as a dev, I misunderstood self-documenting code to mean writing lots of comments explaining everything I wrote. Over time, I realised it’s actually about writing code that speaks for itself—and a big part of that is using clear, meaningful names.

**Reply by steven0x51** [reply]
Clear naming is a big thing, especially in a big team. Every project should have a document about name convention.

**Comment by amankumarrr** [comment]
Consistency in naming conventions is crucial for code readability and maintainability. It ensures that new developers can quickly understand the codebase and helps prevent potential issues that might arise from unclear or inconsistent names. Clear and meaningful naming makes the code more intuitive and easier to work with.

**Comment by sam-s-maher** [comment]
One space I like to really ham up the variable names is for dictionaries - so instead of just 'birthdays' do something like 'birthdateByUserId', and then when iterating on this 'birthdateForUserId'. Would be keen to hear how others like to name theirs?

**Comment by sethdaily** [comment]
Even for a non developer this makes a lot of sense

**Comment by stephanfakossw** [comment]
N/A - I am a developer

I can definitely understand how important this is for devs

**Reply by Tanya-Leahy** [reply]
I am not a Developer, but can understand the need for following proper naming conventions. In previous roles, it was super important to follow a naming convention for consistency. I can see why this would be important also when coding.

**Comment by SamBWagner** [comment]
100%. I like all the suggestions here. The **MOST** important thing here is to keep your names *descriptivec and *consistent*.

If you can't articulate what the variable name means to a grandma in 1 sentence the identifier is likely bad.

**Comment by MattParkerDev** [comment]
This is super important for writing maintainable code.

**Comment by miichaelsmedley** [comment]
I am not a developer but with clear naming conventions as described above it means that someone who is not a developer can make sense of code.

**Comment by jimmidier** [comment]
Naming convention should be semantic and most importantly consistent across the project, for best readability and maintenance.

**Comment by ZenoWang1999** [comment]
One problem I frequently encounter is that many objects we need to name cannot be directly represented as very specific actions or entities. They might be middleware or more abstract concepts. In these cases, we often run into the issue of names becoming too long in an attempt to describe them comprehensively.

**Comment by tino-liu** [comment]
I have seen a function named "DoThis". Imagine the pain... "WHAT IT DO???"
I 100% agree that good naming convention makes everybody's life easier.

**Comment by babakamyljanovssw** [comment]
Having clear and consistent names across the project/company make easier for devs to join the team and understand the project quickly.

**Comment by Lewkans** [comment]
Having good naming conventions means that people reading your code can easily tell what it means, to understand the functionality much easier

**Comment by Aibono1225** [comment]
Well explained in the video, this rule is very important for maintainability and readability

**Comment by ozair-ashfaque** [comment]
Great rule and well explained in the video
